### PR TITLE
marker: init at 2020.04.04

### PR DESCRIPTION
**Pull request by trepetti**
###### Motivation for this change

Add Marker, a GTK3-based markdown editor for Linux desktops.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trepetti** [comment]
Result of `nixpkgs-review pr 100653` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>marker</li>
  </ul>
</details>

File: pkgs/applications/editors/marker/default.nix
Comment (r-burns) [inline pkgs/applications/editors/marker/default.nix:44]:
```suggestion
```
This is implied

File: pkgs/applications/editors/marker/default.nix
Comment (r-burns) [inline pkgs/applications/editors/marker/default.nix:22]: It looks like this is vendoring https://github.com/Mandarancio/scidown. Is marker able to use a system installation of scidown? I think that would be preferable

File: pkgs/applications/editors/marker/default.nix
Comment (r-burns) [inline pkgs/applications/editors/marker/default.nix:49]:
```suggestion
    license = licenses.gpl3Plus;
```
Might want to ask upstream as to whether this is GPL or LGPL - https://github.com/fabiocolacio/Marker/blob/2020.04.04/src/marker.c#L6-L9 says LGPL3+ but the toplevel LICENSE.md is standard GPL.

**Review by r-burns — COMMENTED** [review]
Passes nixpkgs-review on non-nixos linux

File: pkgs/applications/editors/marker/default.nix
Comment (trepetti) [reply pkgs/applications/editors/marker/default.nix:22]: Thank you for your review! Marker's `meson.build` is compiling it all together by explicitly listing all the source files in scidown as opposed to using scidown's build system and then linking after the fact. If I packaged scidown, we would need a pretty big patch for the `meson.build` file. But if you think it is worth it, I can look into it when I get a chance.

File: pkgs/applications/editors/marker/default.nix
Comment (trepetti) [reply pkgs/applications/editors/marker/default.nix:44]: Will fix. Had not realized that, but it definitely makes sense.

File: pkgs/applications/editors/marker/default.nix
Comment (r-burns) [reply pkgs/applications/editors/marker/default.nix:22]: Ah you're right, didn't realize those are all hardcoded. Typically I'd expect a submodule to have a fallback to an external library

File: pkgs/applications/editors/marker/default.nix
Comment (trepetti) [reply pkgs/applications/editors/marker/default.nix:49]: I just opened an [issue](https://github.com/fabiocolacio/Marker/issues/335) and will get to the bottom of this. I didn't actually see any reference to the LGPL in the sources or license file itself, but agree that it is not clear if it is GPL3 or GPL3+.

File: pkgs/applications/editors/marker/default.nix
Comment (trepetti) [reply pkgs/applications/editors/marker/default.nix:22]: Will keep as is then. Yes, you are right that it in general it would be better from a dependency management perspective to have this separate project in Nix as its own package. As you point out, the hardcoded build using the submodules sources is one reason for keeping it as is, and the other is that I do not think scidown is used in many other projects, but who knows, maybe it will be in the future (though there has not been much recent activity in the repository).

File: pkgs/applications/editors/marker/default.nix
Comment (r-burns) [reply pkgs/applications/editors/marker/default.nix:49]: I think the 3+ part is clear, the source files all appear to all say GNU Library General Public License version 3 or later. My confusion is just that the toplevel LICENSE.md is the regular (non-library) type. Thanks for opening the issue, will stay tuned

**Review by r-burns — APPROVED** [review]
Good bot :)

Since upstream hasn't responded I won't try to hold up this PR any longer. I think the most likely explanation is they intended to use LGPL3+ but accidentally copied the GPL license file.  So for now I'll say LGTM 👍

**Review by timokau — APPROVED** [review]
Thanks for the review @r-burns :)

**Comment by alexeymuranov** [comment]
Would it be possible to backport it to 20.09? I can try when i have some spare time...

**Comment by timokau** [comment]
I'm not sure what the policy on backporting new packages is. You can mix packages from different channels though, so you could still install this on a 20.09 setup. Is that good enough for you?

**Comment by alexeymuranov** [comment]
@timokau , it would be easier for me to use everything from one channel, i am not a very advanced user.

According to [this][1] old message of Vladimír Čunát about stabilising NixOS 18.03, backporting new pacakges seems ok:

> What should be cherry-picked?
> - [...]
> - Optionally, new package additions, as that won't break anything. 

[1]: https://groups.google.com/g/nix-devel/c/3KxPNwxDV9E/m/6vZUnhv6BgAJ

**Comment by trepetti** [comment]
[The most recent maintainers wiki entry says we should probably hold off on backporting the package.](https://gist.github.com/grahamc/c60578c6e6928043d29a427361634df6#what-to-backport)

Adding a new channel to nix-env install pacakges is super easy, though:

```
nix-channel --add https://nixos.org/channels/nixpkgs-unstable unstable
nix-env -iA unstable.marker
```

And that should do it. You don't even need to be root.

**Comment by alexeymuranov** [comment]
@trepetti, this didn't work for me:
```console
$ nix-channel --add https://nixos.org/channels/nixpkgs-unstable unstable
$ nix-env -iA unstable.marker                                       ~/

error: attribute 'unstable' in selection path 'unstable.marker' not found
```

**Comment by trepetti** [comment]
Then try `nix-env -i unstable.marker`. You may not be in the `nix.allowedUsers` list. Also make sure you run `nix-channel --update`--forgot to mention that.

**Comment by alexeymuranov** [comment]
@trepetti, yes it was `nix-channel --update` that was missing, thanks. I said i was not a very advanced user :).

**Comment by alexeymuranov** [comment]
@trepetti, the problem with having two channels for me is that now i cannot check for upgraded packages in the stable channel with
```console
$ nix-env -u --dry-run
```
-- it shows everything from the unstable too.

**Comment by trepetti** [comment]
If you are on NixOS, you can have nix-env only upgrade packages using derivations from the default (stable) channel by providing that channel's file:

```
nix-env -u --dry-run -f .nix-defexpr/channels_root/nixos/default.nix
```

**Comment by alexeymuranov** [comment]
@trepetti , thanks. I am on NixOS indeed.

**Comment by timokau** [comment]
> @timokau , it would be easier for me to use everything from one channel, i am not a very advanced user.
> 
> According to [this](https://groups.google.com/g/nix-devel/c/3KxPNwxDV9E/m/6vZUnhv6BgAJ) old message of Vladimír Čunát about stabilising NixOS 18.03, backporting new pacakges seems ok:

That's a good find! Its a little old to base a decision on though. The current guidelines that @alexeymuranov linked don't give any concrete guidance (at least the way I read them). I asked on IRC, but unfortunately didn't get a reply.

@jonringer do you have an opinion on backporting new packages?

**Comment by jonringer** [comment]
I'm okay with it philosophically. But would prefer people to use unstable when they want the latest of something.

**Comment by alexeymuranov** [comment]
@jonringer, in my case i do not want the latest, i want any.

**Pull request merged**
